tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleModuleProjectImporter.java

```diff
@@ -37,49 +37,67 @@ public IStatus canImport( String location )
 
         File file = new File( location );
 
-        if( file.exists() )
+        if( findGradleFile( file ) && findSettingsFile( file ) )
+        {
+            retval = Status.OK_STATUS;
+        }
+        else
         {
-            if( findGradleFile( file ) )
-            {
-                retval = Status.OK_STATUS;
-            }
-
             File parent = file.getParentFile();
 
             while( parent != null )
             {
-                if( findGradleFile( parent ) )
+                if( findGradleFile( parent ) && findSettingsFile( file ) )
                 {
```

Comment:
Hey @gamerson  I didn't get you mean . If the loaction has a settings.gradle and one of it's ancestors has a build.gradle and it is ok ? Based on your code , I think it is ok if the location has both build.gradle and settings.gradle.